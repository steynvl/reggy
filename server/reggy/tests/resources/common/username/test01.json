{
  "type": "Username",
  "information": {
    "shouldStartWith": "Anything",
    "shouldContain": [
      "Lowercase letter",
      "Special character"
    ],
    "minimumLength": "2",
    "maximumLength": "7"
  },
  "generalRegexInfo": {
    "startRegexMatchAt": "Anywhere",
    "endRegexMatchAt": "Anywhere",
    "regexTarget": "Java"
  },
  "generateMethod": "commonUseCases"
}
