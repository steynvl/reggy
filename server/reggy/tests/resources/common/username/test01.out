Java
Pattern regex = Pattern.compile("(?=.*[a-z])(?=.*[!\"#$%&'()*+,-.:;<=>?@\\[\\]^_`{|}~]).{2,7}");
(?=.*[a-z])(?=.*[!\"#$%&'()*+,-.:;<=>?@\\[\\]^_`{|}~]).{2,7}
Perl
my $regex = /(?=.*[a-z])(?=.*[!"#$%&'()*+,-.:;<=>?@\[\]^_`{|}~]).{2,7}/;
(?=.*[a-z])(?=.*[!"#$%&'()*+,-.:;<=>?@\[\]^_`{|}~]).{2,7}
POSIX
(?=.*[a-z])(?=.*[!"#$%&'()*+,-.:;<=>?@\[\]^_`{|}~]).{2,7}
(?=.*[a-z])(?=.*[!"#$%&'()*+,-.:;<=>?@\[\]^_`{|}~]).{2,7}
Python
regex = re.compile(r'(?=.*[a-z])(?=.*[!"#$%&\'()*+,-.:;<=>?@\[\]^_`{|}~]).{2,7}')
(?=.*[a-z])(?=.*[!"#$%&\'()*+,-.:;<=>?@\[\]^_`{|}~]).{2,7}
JavaScript
const regex = /(?=.*[a-z])(?=.*[!"#$%&'()*+,-.:;<=>?@\[\]^_`{|}~]).{2,7}/;
(?=.*[a-z])(?=.*[!"#$%&'()*+,-.:;<=>?@\[\]^_`{|}~]).{2,7}
PHP
$regex = '/(?=.*[a-z])(?=.*[!"#$%&'()*+,-.:;<=>?@\[\]^_`{|}~]).{2,7}/';
(?=.*[a-z])(?=.*[!"#$%&'()*+,-.:;<=>?@\[\]^_`{|}~]).{2,7}
Golang
regex, _ := regexp.Compile("(?=.*[a-z])(?=.*[!\"#$%&'()*+,-.:;<=>?@\\[\\]^_`{|}~]).{2,7}")
(?=.*[a-z])(?=.*[!\"#$%&'()*+,-.:;<=>?@\\[\\]^_`{|}~]).{2,7}
Rust
let re = Regex::new(r"(?=.*[a-z])(?=.*[!"#$%&\'()*+,-.:;<=>?@\[\]^_`{|}~]).{2,7}").unwrap();
(?=.*[a-z])(?=.*[!"#$%&\'()*+,-.:;<=>?@\[\]^_`{|}~]).{2,7}
C#
var re = new Regex(@"(?=.*[a-z])(?=.*[!"#$%&\'()*+,-.:;<=>?@\[\]^_`{|}~]).{2,7}");
(?=.*[a-z])(?=.*[!"#$%&\'()*+,-.:;<=>?@\[\]^_`{|}~]).{2,7}
Scala
val re = "(?=.*[a-z])(?=.*[!\"#$%&'()*+,-.:;<=>?@\\[\\]^_`{|}~]).{2,7}".r
(?=.*[a-z])(?=.*[!\"#$%&'()*+,-.:;<=>?@\\[\\]^_`{|}~]).{2,7}
Kotlin
val regex = Regex("(?=.*[a-z])(?=.*[!\"#$%&'()*+,-.:;<=>?@\\[\\]^_`{|}~]).{2,7}")
(?=.*[a-z])(?=.*[!\"#$%&'()*+,-.:;<=>?@\\[\\]^_`{|}~]).{2,7}